extras/subscript.go

```diff
@@ -0,0 +1,115 @@
+package extras
+
+import (
+	"github.com/gohugoio/hugo-goldmark-extensions/extras/ast"
+	"github.com/yuin/goldmark"
+	gast "github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/renderer/html"
+	"github.com/yuin/goldmark/text"
+	"github.com/yuin/goldmark/util"
+)
+
+type subscriptDelimiterProcessor struct {
+}
+
+func (p *subscriptDelimiterProcessor) IsDelimiter(b byte) bool {
+	return b == '~'
```

Comment:
This is a general comment that spans all insert/subscript/superscript.go files:

I have skimmed the implementation of all of these 3 and they look almost exactly the same, the only diff seems to be the delimiter and the HTML wrapper element.

I don't mind code duplication, but in this case I think we can make everything simpler by wrapping all of these into one extension/parser/renderer.